homeassistant/components/tautulli/config_flow.py

```diff
@@ -0,0 +1,140 @@
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import (
+    DEFAULT_NAME,
+    DEFAULT_PATH,
+    DEFAULT_PORT,
+    DEFAULT_SSL,
+    DEFAULT_VERIFY_SSL,
+    DOMAIN,
+)
+
+
+class TautulliConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Tautulli."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initiated by the user."""
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+        errors = {}
+        if user_input is not None:
+            error = await self.validate_input(user_input)
+            if error is None:
+                return self.async_create_entry(
+                    title=DEFAULT_NAME,
+                    data=user_input,
+                )
+            errors["base"] = error
+
+        user_input = user_input or {}
+        data_schema = {
+            vol.Required(CONF_API_KEY, default=user_input.get(CONF_API_KEY, "")): str,
+            vol.Required(CONF_URL, default=user_input.get(CONF_URL, "")): str,
+            vol.Optional(
+                CONF_VERIFY_SSL, default=user_input.get(CONF_VERIFY_SSL, True)
+            ): bool,
+        }
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(data_schema),
+            errors=errors or {},
+        )
+
+    async def async_step_reauth(self, config: dict[str, Any]) -> FlowResult:
+        """Handle a reauthorization flow request."""
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, str] | None = None
+    ) -> FlowResult:
+        """Confirm reauth dialog."""
+        errors = {}
+        entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+        if user_input is not None and entry:
+            _input = {**entry.data, CONF_API_KEY: user_input[CONF_API_KEY]}
+            error = await self.validate_input(_input)
+            if error is None:
+                self.hass.config_entries.async_update_entry(entry, data=_input)
+                await self.hass.config_entries.async_reload(entry.entry_id)
+                return self.async_abort(reason="reauth_successful")
+            errors["base"] = error
+        return self.async_show_form(
+            step_id="reauth_confirm",
+            data_schema=vol.Schema({vol.Required(CONF_API_KEY): str}),
+            errors=errors,
+        )
+
+    async def async_step_import(self, config: dict[str, Any]) -> FlowResult:
+        """Import a config entry from configuration.yaml."""
+        if self._async_current_entries():
+            _LOGGER.warning(
+                "Tautulli config with host %s import failed. Only one instance is supported",
+                config[CONF_HOST],
+            )
```

Comment:
You've already got a message in the platform setup, making this a duplicate

```suggestion
```

Reply:
Must have forgotten to remove this when asked to move it to `sensor.py`.